tgui/packages/tgui/interfaces/TurbineComputer.js

```diff
@@ -0,0 +1,57 @@
+import { useBackend } from '../backend';
+import { Button, LabeledList, Section} from '../components';
+import { Window } from '../layouts';
+
+export const TurbineComputer = (props, context) => {
+  const { act, data } = useBackend(context);
+  const operational = Boolean(data.compressor && !data.compressor_broken && data.turbine && !data.turbine_broken);
+  return (
+    <Window>
+      <Window.Content>
+        <Section title="Status" buttons={(
+            <>
+              <Button
+                icon={data.online ? 'power-off' : 'times'}
+                content={data.online ? 'Online' : 'Offline'}
+                selected={data.online}
+                disabled={!operational}
+                onClick={() => act('toggle_power')}
+              />
+              <Button
+                icon="sync"
+                content="Reconnect"
+                onClick={() => act('reconnect')}
+              />
+            </>
+          )}>
+          {!operational && (
+            <LabeledList>
+              <LabeledList.Item
+                label="Compressor Status"
+                color={(!data.compressor || data.compressor_broken) ? 'bad' : 'good'}>
+                {data.compressor_broken? data.compressor ? 'Offline' : 'Missing' : 'Online'}
+              </LabeledList.Item>
+              <LabeledList.Item
+                label="Turbine Status"
+                color={(!data.turbine || data.turbine_broken) ? 'bad' : 'good'}>
+                {data.turbine_broken ? data.turbine ? 'Offline' : 'Missing' : 'Online'}
+              </LabeledList.Item>
+            </LabeledList>
+          ) || (
+            <LabeledList>
+              <LabeledList.Item label="Turbine Speed">
+                {data.rpm} RPM
+              </LabeledList.Item>
+              <LabeledList.Item label="Internal Temp">
+                {data.temperature} K
+              </LabeledList.Item>
+              <LabeledList.Item label="Generated Power">
+                {data.power} W
+              </LabeledList.Item>
+            </LabeledList>
```

Comment:
just split these two sections off into their own components consts to clean this up and separate the UI elements from the rendering logic

Reply:
Updated on commit [a289229](https://github.com/ParadiseSS13/Paradise/pull/21226/commits/a28922997d3a693fe4e8bb7ce63e242ded01e0fe). Unfortunately the reconnect button does not function as intended now when building a new or rebuilding a turbine.